Storage/src/StorageClient.php

```diff
@@ -425,4 +416,127 @@ public function getServiceAccount(array $options = [])
+     * // Get the HMAC keys associated with a Service Account email
+     * $hmacKeys = $storage->hmacKeys([
+     *     'serviceAccountEmail' => $serviceAccountEmail
+     * ]);
+     * ```
+     *
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $serviceAccountEmail If present, only keys for the given
+     *           service account are returned.
+     *     @type bool $showDeletedKeys Whether or not to show keys in the
+     *           DELETED state.
+     *     @type string $userProject If set, this is the ID of the project which
+     *           will be billed for the request.
+     * }
+     * @return ItemIterator<HmacKey>
+     */
+    public function hmacKeys(array $options = [])
+    {
+        $this->requireProjectId();
+
+        $resultLimit = $this->pluck('resultLimit', $options, false);
+        return new ItemIterator(
+            new PageIterator(
+                function (array $metadata) {
+                    return $this->hmacKey(
+                        $metadata['accessId'],
+                        $metadata
+                    );
+                },
+                [$this->connection, 'listHmacKeys'],
+                ['projectId' => $this->projectId] + $options,
+                ['resultLimit' => $resultLimit]
+            )
+        );
+    }
+
+    /**
+     * Lazily instantiate an HMAC Key instance using an Access ID.
+     *
+     * Example:
+     * ```
+     * $hmacKey = $storage->hmacKey($accessId);
+     * ```
+     *
+     * @param string $accessId The ID of the HMAC Key.
+     * @param array $metadata [optional] HMAC key metadata.
+     * @return HmacKey
+     */
+    public function hmacKey($accessId, array $metadata = [])
+    {
+        $this->requireProjectId();
+
+        return new HmacKey($this->connection, $this->projectId, $accessId, $metadata);
+    }
+
+    /**
+     * Creates a new HMAC key for the specified service account.
+     *
+     * Please note that the HMAC secret is only available at creation. Make sure
+     * to note the secret after creation.
+     *
+     * Example:
+     * ```
+     * $response = $storage->createHmacKey('account@myProject.iam.gserviceaccount.com');
+     * $secret = $response->secret();
+     * ```
+     *
+     * @param string $serviceAccountEmail Email address of the service account.
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $userProject If set, this is the ID of the project which
+     *           will be billed for the request. **NOTE**: This option is
+     *           currently ignored by Cloud Storage.
+     * }
+     * @return CreatedHmacKey
+     */
+    public function createHmacKey($serviceAccountEmail, array $options = [])
+    {
+        $this->requireProjectId();
+
+        $res = $this->connection->createHmacKey([
+            'projectId' => $this->projectId,
+            'serviceAccountEmail' => $serviceAccountEmail
+        ] + $options);
```

Comment:
@jdpedrie is it possible to overwrite the projectId using options making it a mutable value for each hmac operation?

Reply:
No, values in `$options` will not override those specified in the first array:

```
php > var_dump(['projectId' => 'a'] + ['projectId' => 'b']);
php shell code:1:
array(1) {
  'projectId' =>
  string(1) "a"
}
```

Reply:
Could you make Project ID configurable for HMAC methods? By default it should use the client specified project Id.

Reply:
@frankyn done.